mongomock/aggregate.py

```diff
@@ -625,6 +626,24 @@ def _handle_conditional_operator(self, operator, values):
             'aggregation pipeline, it is currently not implemented '
             ' in Mongomock.' % operator)
 
+    def _handle_control_flow_operator(self, operator, values):
+        if operator == '$switch':
+            branches = values['branches']
+            default = values.get('default')
+            for branch in branches:
+                try:
+                    case = self.parse(branch['case'])
+                except KeyError:
+                    case = False
```

Comment:
Please add a test to cover that case (when the `case` key is missing).

Reply:
Looking at the [documentation](https://docs.mongodb.com/manual/reference/operator/aggregation/switch/#:~:text=must%20contain%20only%20a%20case%20field) the `case` is not optional:

> The objects in the branches array **must** contain only a `case` field and a `then` field.

Here I am trying to deal with the situation that the value does not exists, like this one already added to tests:
```py
{'case': '$missing_field', 'then': 'first case'},
```

I run the tests on real MongoDB instance and the missing field in `case` jumps to next branch, but missing field in `then` skips entire `$switch`. 

I will extract `branch['case']` before `try-except` to show that code needs to handle only the exceptions raised by `self.parse(...)` and improve the tests.